### PR TITLE
DB-9503 Use subsplitting for analyze when mergeStats=true

### PR DESCRIPTION
**Pull request by yxia92**

### CODE REVIEW

**Comment by yxia92** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED

**Comment by yxia92** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED

File: splice_machine/src/main/java/com/splicemachine/derby/utils/StatisticsAdmin.java [hunk 1]
```diff
@@ -1140,6 +1152,28 @@ public ExecRow next() {
...
+            throw StandardException.plainWrapException(ioe);
+        } finally {
+            try {
+                if (root != null)
+                    root.close();
+            } catch (IOException e) {
+                throw StandardException.plainWrapException(e);
+            }
+        }
+    }
```
Comment (dgomezferro) [inline splice_machine/src/main/java/com/splicemachine/derby/utils/StatisticsAdmin.java:1175]: I think you could use here a try-with-resources here to simplify the code a bit:
```
try (Partition root = SIDriver.driver().getTableFactory().getTable(table)) {
        return root != null ?  root.subPartitions(true).size() : 1;
} catch (Exception ioe) {
        throw StandardException.plainWrapException(ioe);
}
```
If I'm not mistaken the catch block will also catch an exception on root.close()

Comment (yxia92) [reply splice_machine/src/main/java/com/splicemachine/derby/utils/StatisticsAdmin.java]: Thanks @dgomezferro ! I've made the simplification as suggested.

**Comment by yxia92** [comment]
jenkins please test branch @cdh5.12.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**